### DISCUSSION TITLE
How to setup wiki.js cluster or sync wiki.js instances in different network

### DISCUSSION THREAD

**Discussion by biganan2012** [requarks/wiki#4361] [Help / Questions]
Hi Team,

Happy Monday.
I am thinking about the solution to have wiki.js sync up contents in different networks. Please check the details below:

Background:
1, We have 2 networks. One for UAT and another for Prod. Basically there is no connection between UAT and Prod. But as far as I know, we can raise request like NAT to create connections.
2, We are going to deploy 2 instances in UAT and Prod. What's more, we need to keep the contents the same, or 5 minutes delay. 
3, We can write in UAT but read only in Prod. I am mentioning this just to make it easy to single sync instead of sync up each other. 

Solution:
1, At first, i think gitlab might be a good  choice. We can have UAT to push backup to gitlab, and Prod to pull from gitlab. The problem is gitlab can only be accessed from just one environment, UAT for Prod. I am thinking about this because git support both push and pull. May be we can consider some network operations to make gitlab works for both environment? 
2, For file system, it is not easy as git. Because for file storage, wiki.js only support push but not pull. There should be manual works in Prod if backup file of UAT could be transfered to Prod. Not sure if NAT and SFTP can do it.
3, For DB, it might be like this. UAT dump backup files and transfer to Prod, then Prod can import DB files. I can write scripts to automate this if network is not a problem.
4, Use same DB, but not a single DB. We can setup PorstgreSQL cluster, master node in UAT and slave node in Prod? I am not sure if nodes could communicate with each other by NAT.
5, Use K8s? Really confused about the communications between networks.

Need Help:
1, Do we have cluster setting on the branch? Or we may have it in nealy future?
2, Based on my background, could someone give me a suggestion please? Really appreciate.

**Comment by pping** [comment]
Prod git pull from UAT

**Reply by biganan2012** [reply]
Thank you for your reply. I just confused about user/group information. Could these information be synced up with Git please? I have tried Git sync, but seems only contents got synced.
